qutebrowser/misc/sessions.py

```diff
@@ -142,6 +142,11 @@ def _save_tab(self, tab, active):
                 'url': bytes(item.url().toEncoded()).decode('ascii'),
                 'title': item.title(),
             }
+            if len(item_data['title']) == 0 and idx == 0: #Most recent first
+                item_data['title'] = tab.title()
+            if len(item_data['title']) == 0:
```

Comment:
You'd usually just do `if not item_data['title']:` instead of `if len(item_data['title']) == 0`, as empty strings/dicts/lists/... are falsey in Python.

I can fix it up if you prefer, just thought I'd let you know :wink: